src/action/finder/request.rs

```diff
@@ -0,0 +1,1309 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! Defines an internal type for client side file finder action and provides
+//! a function converting proto format of the request
+//! `rrg_proto::FileFinderArgs` to the internal format.
+
+use crate::session::{
+    parse_enum, time_from_micros, ParseError, ProtoEnum, RegexParseError,
+};
+use regex::Regex;
+use rrg_proto::{
+    FileFinderAccessTimeCondition, FileFinderAction, FileFinderArgs,
+    FileFinderCondition, FileFinderContentsLiteralMatchCondition,
+    FileFinderContentsRegexMatchCondition, FileFinderDownloadActionOptions,
+    FileFinderExtFlagsCondition, FileFinderHashActionOptions,
+    FileFinderInodeChangeTimeCondition, FileFinderModificationTimeCondition,
+    FileFinderSizeCondition, FileFinderStatActionOptions,
+};
+use std::convert::TryFrom;
+
+type HashActionOversizedFilePolicy =
+    rrg_proto::file_finder_hash_action_options::OversizedFilePolicy;
+type DownloadActionOversizedFilePolicy =
+    rrg_proto::file_finder_download_action_options::OversizedFilePolicy;
+type RegexMatchMode =
+    rrg_proto::file_finder_contents_regex_match_condition::Mode;
+type LiteralMatchMode =
+    rrg_proto::file_finder_contents_literal_match_condition::Mode;
+type ActionType = rrg_proto::file_finder_action::Action;
+type ConditionType = rrg_proto::file_finder_condition::Type;
+type XDevMode = rrg_proto::file_finder_args::XDev;
+type PathType = rrg_proto::path_spec::PathType;
+
+#[derive(Debug)]
+pub struct Request {
+    pub paths: Vec<String>,
```

Comment:
There is a `Path` type that should be used to represent paths.

Reply:
It's not really a path in this place - it can contain alternatives (e.g. "/home/{a,b,c/d,..}/test"), so it's simpler to use string to process it in early stages. I'm changing it to PathBuf after the preprocessing part.

Reply:
Then you have an incorrectly named field.

Reply:
This field is a copy of FileFinderArgs::paths (which has the same type). I think that renaming it wouldn't be helpful, because idea of this field is being a `path`.